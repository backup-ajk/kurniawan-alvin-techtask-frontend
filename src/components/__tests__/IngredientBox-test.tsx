import React from 'react';
import renderer from 'react-test-renderer';

import { IngredientBox } from '../';

it('should render Ingredient Box successfully', () => {
  const ingredient = {
    title: 'Chicken',
    useBy: '2020-12-30',
  };
  const mockFunction = jest.fn();
  const iBox = renderer
    .create(
      <IngredientBox
        title={ingredient.title}
        useBy={ingredient.useBy}
        selected={false}
        disabled={false}
        onClick={mockFunction}
      />,
    )
    .toJSON();
  expect(iBox).toMatchSnapshot();
});
