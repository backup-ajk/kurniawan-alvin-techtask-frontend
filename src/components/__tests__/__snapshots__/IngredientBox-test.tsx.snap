// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Ingredient Box successfully 1`] = `
<div
  onClick={[Function]}
  style={
    Object {
      "borderColor": "#bcbfff",
      "borderRadius": 10,
      "borderStyle": "solid",
      "borderWidth": 2,
      "cursor": "pointer",
      "display": "flex",
      "flexDirection": "column",
      "height": 90,
      "position": "relative",
    }
  }
>
  <div
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#dfecff",
        "borderTopLeftRadius": 8,
        "borderTopRightRadius": 8,
        "display": "flex",
        "flex": 1,
        "justifyContent": "center",
        "padding": 10,
      }
    }
  >
    Chicken
  </div>
  <div
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#92c0ff",
        "borderBottomLeftRadius": 8,
        "borderBottomRightRadius": 8,
        "display": "flex",
        "height": 30,
        "justifyContent": "center",
      }
    }
  >
    30-12-2020
  </div>
</div>
`;
